viz_scripts/survey_responses.ipynb

```diff
@@ -0,0 +1,403 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c751b118",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#create translation dictionaries\n",
+    "opt_dict, quest_dict = build_dictionaries(sheet_list)\n",
+    "print(\"Questions dictionary:\\n\",quest_dict)\n",
+    "print(\"Options dictionary:\\n\", opt_dict)\n",
+    "\n",
+    "color_map = mapping_color_surveys(opt_dict)\n",
+    "\n",
+    "#format survey trips into responses dataframe\n",
+    "if len(survey_trips) > 0:\n",
+    "    df_responses = create_dataframe(survey_trips)\n",
+    "else:\n",
+    "    df_responses = survey_trips.copy()\n",
+    "    \n",
+    "    \n",
+    "file_suffix = scaffolding.get_file_suffix(year, month, program)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "93dbaa5e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# #create the total dfs\n",
+    "# total_dfs = {}\n",
+    "# #for conditional surveys!\n",
+    "# wrapped_config = {'survey_info': survey_info}\n",
+    "# if 'buttons' in survey_info.keys():\n",
+    "#     all_confirmed_trips = all_confirmed_trips[all_confirmed_trips['ble_sensed_summary'].notna()]\n",
+    "#     all_confirmed_trips[\"confirmedMode_baseMode\"] = all_confirmed_trips.ble_sensed_summary.apply(lambda md: max(md[\"distance\"], key=md[\"distance\"].get))\n",
+    "#     all_confirmed_trips['survey_name_prompted'] = all_confirmed_trips.apply(lambda row: conditional_surveys.survey_prompted_for_trip(row.to_dict(), wrapped_config), axis=1)\n",
+    "    \n",
+    "#     for survey_name in list(sheet_list.keys()):\n",
+    "#         if survey_name in all_confirmed_trips['survey_name_prompted'].unique():\n",
+    "#             total_dfs[survey_name] = all_confirmed_trips[all_confirmed_trips['survey_name_prompted'] == survey_name]\n",
+    "#         else:\n",
+    "#             #never prompted\n",
+    "#             total_dfs[survey_name] = pd.DataFrame()\n",
+    "# else:\n",
+    "#     survey_name = list(sheet_list.keys())[0] #there is only one if non-conditional\n",
+    "#     total_dfs[survey_name] = all_confirmed_trips\n",
+    "\n",
+    "# for key in total_dfs.keys():\n",
+    "#     if len(total_dfs[key]) > 0:\n",
+    "#         print(key, \":\", len(total_dfs[key]), \"trips\", total_dfs[key].user_id.nunique(), \"users\")\n",
+    "#     else:\n",
+    "#         print(key, \":\", len(total_dfs[key]), \"trips\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5732576d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#merge any cols with the same name into 1 col -- should have different values in their survey_name col\n",
+    "#https://stackoverflow.com/questions/24390645/python-pandas-merge-samed-name-columns-in-a-dataframe\n",
+    "def sjoin(x): return ';'.join(x[x.notnull()].astype(str))\n",
+    "df_responses = df_responses.groupby(level=0, axis=1).apply(lambda x: x.apply(sjoin, axis=1))"
```

Comment:
what is this for? If the responses are in different surveys, should they be combined into a single column?

Reply:
I see this was in https://github.com/e-mission/em-public-dashboard/pull/124#issuecomment-2094657673 but I don't fully understand why the column is duplicated. Shouldn't the survey id be part of the column name as well?

Reply:
As it stands, the survey id is not part of the column name - I could refactor to include it as a part of the column name and update the way that I translate the column names to be the chart titles

They should end up in different charts since we group and filter by the `survey_name`, which is its own column